cla_backend/apps/cla_eventlog/management/commands/find_and_delete_old_cases.py

```diff
@@ -0,0 +1,31 @@
+from django.core.management.base import BaseCommand
+from dateutil.relativedelta import relativedelta
+
+from legalaid.models import Case
+from cla_butler.tasks import DeleteOldData
+from cla_eventlog.models import Log
+
+
+class FindAndDeleteCasesUsingCreationTime(DeleteOldData):
+    def get_eligible_cases(self):
+        two_years = self.now - relativedelta(years=2)
+        old_cases = []
+
+        filtered_cases = Case.objects.filter(created__lte=two_years)
+        for case in filtered_cases:
+            try:
+                latest_log = Log.objects.filter(case__id=case.id).latest("created")
+                if latest_log.created <= two_years:
+                    old_cases.append(case.id)
+            except Log.DoesNotExist:
+                # Old case without any event logs
+                old_cases.append(case.id)
```

Comment:
Could this be combined into the filtering of the `Case` queryset?

Reply:
No because we need access to logs and there is no direct relationship between log and case

Reply:
The Django ORM lets you filter across ManyToMany relationships